java/java.mx.project/src/org/netbeans/modules/java/mx/project/SuiteSources.java

```diff
@@ -0,0 +1,1195 @@
+                if (libs.size() == 1) {
+                    prjCp = ClassPathSupport.createProxyClassPath(prjCp,
+                                                                  ClassPathFactory.createClassPath(libs.get(0))
+                    );
+                } else {
+                    prjCp = ClassPathSupport.createProxyClassPath(prjCp,
+                                                                  ClassPathFactory.createClassPath(
+                                                                                  ClassPathSupport.createProxyClassPathImplementation(
+                                                                                                  libs.toArray(new ClassPathImplementation[0])
+                                                                                  )
+                                                                  )
+                    );
+                }
+            }
+            return prjCp;
+        }
+
+        ClassPath getProcessorCP() {
+            computeTransitiveDeps();
+            return processorPath;
+        }
+
+        @Override
+        public Set<? extends AnnotationProcessingQuery.Trigger> annotationProcessingEnabled() {
+            return EnumSet.of(AnnotationProcessingQuery.Trigger.ON_SCAN, AnnotationProcessingQuery.Trigger.IN_EDITOR);
+        }
+
+        @Override
+        public Iterable<? extends String> annotationProcessorsToRun() {
+            return null;
+        }
+
+        @Override
+        public URL sourceOutputDirectory() {
+            return srcGenDir == null ? null : srcGenDir.toURL();
+        }
+
+        @Override
+        public Map<? extends String, ? extends String> processorOptions() {
+            return Collections.emptyMap();
+        }
+
+        @Override
+        public void addChangeListener(ChangeListener l) {
+        }
+
+        @Override
+        public void removeChangeListener(ChangeListener l) {
+        }
+
+        @Override
+        public SuiteSources owner() {
+            return SuiteSources.this;
+        }
+    }
+
+    private class Library implements FlaggedClassPathImplementation, Dep {
+        final MxLibrary lib;
+        final PropertyChangeSupport support = new PropertyChangeSupport(this);
+        final String libName;
+        Collection<Dep> allDeps;
+        Boolean exists;
+
+        Library(String libName, MxLibrary lib) {
+            this.libName = libName;
+            this.lib = getOSSLibrary(lib);
+        }
+
+        final MxLibrary getOSSLibrary(MxLibrary lib) {
+            if (lib.sha1() == null && !lib.os_arch().isEmpty()) {
+                Map<String, MxLibrary.Arch> os_dep_libs = lib.os_arch();
+                String os = System.getProperty("os.name").toLowerCase();
+                for (Map.Entry<String, MxLibrary.Arch> entry : os_dep_libs.entrySet()) {
+                    if (os.contains(entry.getKey())) {
+                        return entry.getValue().amd64();
```

Comment:
Uh :) I meant `mx` already supports `aarch64`, maybe even `i386` ?

Reply:
This silently assumes all developer laptops and workstations are AMD64. True for now, until Apple own processor becomes popular among devs. I guess it is going to take a bit of time and it will need deeper fixes in `mx` itself then.